src/Tizen.Peripheral/Tizen.Peripheral/I2c.cs

```diff
@@ -0,0 +1,151 @@
+                throw ExceptionFactory.CreateException(ret);
+
+            return data;
+        }
+
+        /// <summary>
+        /// Writes the bytes data to the I2C slave device.
+        /// </summary>
+        /// <param name="data"></param>
+        public void Write(byte[] data)
+        {
+            var length = Convert.ToUInt32(data.Length);
+            var ret = NativeI2c.Write(_handle, data, length);
+            if (ret != Internals.Errors.ErrorCode.None)
+                throw ExceptionFactory.CreateException(ret);
+        }
+
+        /// <summary>
+        /// Reads single byte data from the register of the I2C slave device.
+        /// </summary>
+        /// <param name="register">The register address of the I2C slave device to read.</param>
+        /// <returns>The single byte data.</returns>
+        public byte ReadRegisterByte(byte register)
+        {
+            var ret = NativeI2c.ReadRegisterByte(_handle, register, out byte data);
+            if (ret != Internals.Errors.ErrorCode.None)
+                throw ExceptionFactory.CreateException(ret);
+
+            return data;
+        }
+
+        /// <summary>
+        /// Writes single byte data to the register of the I2C slave device.
+        /// </summary>
+        /// <param name="register">The register address of the I2C slave device to write.</param>
+        /// <param name="data">The single byte data to write.</param>
+        public void WriteRegisterByte(byte register, byte data)
+        {
+            var ret = NativeI2c.WriteRegisterByte(_handle, register, data);
+            if (ret != Internals.Errors.ErrorCode.None)
+                throw ExceptionFactory.CreateException(ret);
+        }
+
+        /// <summary>
+        /// Reads word data from the register of the I2C slave device.
+        /// </summary>
+        /// <param name="register">The register address of the I2C slave device to read.</param>
+        /// <returns>The word (2 bytes) data.</returns>
+        public ushort ReadRegisterWord(byte register)
+        {
+            var ret = NativeI2c.ReadRegisterWord(_handle, register, out ushort data);
+            if (ret != Internals.Errors.ErrorCode.None)
+                throw ExceptionFactory.CreateException(ret);
+
+            return data;
+        }
+
+        /// <summary>
+        /// Writes word data to the register of the I2C slave device.
+        /// </summary>
+        /// <param name="register">The register address of the I2C slave device to write.</param>
+        /// <param name="data">The word (2 bytes) data to write.</param>
+        public void WriteRegisterWord(byte register, ushort data)
+        {
+            var ret = NativeI2c.WriteRegisterWord(_handle, register, data);
+            if (ret != Internals.Errors.ErrorCode.None)
+                throw ExceptionFactory.CreateException(ret);
+        }
+
+        /// <summary>
+        /// Disposes the I2c.
+        /// </summary>
+        public void Dispose()
```

Comment:
Implement the dispose pattern
https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose